gunicorn/arbiter.py

```diff
@@ -402,10 +414,15 @@ def reexec(self):
         self.cfg.pre_exec(self)
 
         environ = self.cfg.env_orig.copy()
-        fds = [l.fileno() for l in self.LISTENERS]
-        environ['GUNICORN_FD'] = ",".join([str(fd) for fd in fds])
         environ['GUNICORN_PID'] = str(master_pid)
 
+        if self.systemd:
+            environ['LISTEN_PID'] = str(os.getpid())
+            environ['LISTEN_FDS'] = str(len(self.LISTENERS))
+        else:
+            fds = [l.fileno() for l in self.LISTENERS]
+            environ['GUNICORN_FD'] = ",".join([str(fd) for fd in fds])
```

Comment:
Since `fds` is not used anywhere else, perhaps we can change it to:

```py
environ['GUNICORN_FD'] = ",".join(str(l.fileno()) for l in self.LISTENERS)
```